Bio/SeqIO/GfaIO.py

```diff
@@ -0,0 +1,89 @@
+"""Bio.SeqIO support for the Graphical Fragment Assembly format."""
+
+import warnings
+import hashlib
+
+from Bio import BiopythonWarning
+from Bio.File import as_handle
+from Bio.Seq import Seq
+from Bio.SeqRecord import SeqRecord
+
+
+def _check_tags(seq, tags):
+    """Check a segment line's tags for inconsistencies (PRIVATE)."""
+    for tag in tags:
+        if tag[:2] == "LN":
+            # Sequence length
+            if int(tag[5:]) != len(seq):
+                warnings.warn(
+                    f"Segment line has incorrect length. Expected {tag[5:]} but got {len(seq)}.",
+                    BiopythonWarning,
+                )
+        if tag[:2] == "SH":
+            # SHA256 checksum
+            checksum = hashlib.sha256(str(seq).encode()).hexdigest()
+            if checksum.upper() != tag[5:]:
+                warnings.warn(
+                    f"Segment line has incorrect checksum. Expected {tag[5:]} but got {checksum}.",
+                    BiopythonWarning,
+                )
+
+
+def GfaIterator(source):
+    """Parser for GFA 1.x files.
+
+    Documentation: https://gfa-spec.github.io/GFA-spec/GFA1.html
+    """
+    with as_handle(source) as handle:
+        for line in handle:
+            if line == "\n":
+                warnings.warn("GFA data has a blank line.", BiopythonWarning)
+
+            fields = line.strip("\n").split("\t")
+            if fields[0] != "S":
+                continue
+            if len(fields) < 3:
+                raise ValueError(
+                    f"Segment line must have name and sequence fields: {line}."
+                )
+
+            if fields[2] == "*":
+                # no sequence data
```

Comment:
We might still want to yield a record with unknown sequence? Are these common in real GFA output?